### PR TITLE
Make Kinesis update interval from S3 configurable

### LINKED ISSUE #1609: Make Presto Kinesis metadata update interval from S3 configurable
Check startS3Updates method in S3TableConfigClient.java

### PR DESCRIPTION
**Pull request by somayaj**
Fixes #1609 
Supersedes #6455

### CODE REVIEW

File: plugin/trino-kinesis/src/test/java/io/trino/plugin/kinesis/TestKinesisConfig.java [hunk 1]
```diff
@@ -77,6 +79,7 @@ public void testExplicitPropertyMappings()
                 .setTableDescriptionLocation("/var/lib/kinesis")
                 .setDefaultSchema("kinesis")
                 .setHideInternalColumns(false)
+                .setUpdateInterval(new Duration(1, TimeUnit.MINUTES))
```
Comment (ebyhr) [inline plugin/trino-kinesis/src/test/java/io/trino/plugin/kinesis/TestKinesisConfig.java:82]: This test will fail because the value is 600000ms. How about smaller one `500m` and `500, TimeUnit.MILLISECONDS`?

File: plugin/trino-kinesis/src/main/java/io/trino/plugin/kinesis/KinesisConfig.java [hunk 2]
```diff
@@ -28,6 +29,7 @@
     private String defaultSchema = "default";
     private String tableDescriptionLocation = "etc/kinesis/";
     private boolean hideInternalColumns = true;
+    private Duration updateInterval = new Duration(1, TimeUnit.MINUTES);
```
Comment (ebyhr) [inline plugin/trino-kinesis/src/main/java/io/trino/plugin/kinesis/KinesisConfig.java:32]: Let's rename to `tableDescriptionRefreshInterval` and the relevant places. Also, move to directly under `tableDescriptionLocation`.

**Review by ebyhr — COMMENTED** [review]
Thanks for revising this commit. Please update https://github.com/trinodb/trino/blob/master/docs/src/main/sphinx/connector/kinesis.rst as well.

**Comment by somayaj** [comment]
> Thanks for revising this commit. Please update https://github.com/trinodb/trino/blob/master/docs/src/main/sphinx/connector/kinesis.rst as well.

Hi, can you work this? I don't think I have enough detail to add here to the docs. Thank you.

**Comment by ebyhr** [comment]
@somayaj No need to think hard. Simple description likes `@ConfigDescription` is fine.

**Comment by somayaj** [comment]
ok, thanks

**Comment by somayaj** [comment]
Hi, @ebyhr are we good with these changes?

File: plugin/trino-kinesis/src/test/java/io/trino/plugin/kinesis/TestKinesisConfig.java [hunk 3]
```diff
@@ -43,7 +44,7 @@ public void testDefaults()
                 .setCheckpointEnabled(false)
                 .setDynamoReadCapacity(50)
                 .setDynamoWriteCapacity(10)
-                .setCheckpointInterval(new Duration(60000, TimeUnit.MILLISECONDS))
+                .setCheckpointInterval(new Duration(500, TimeUnit.MILLISECONDS))
```
Comment (ebyhr) [inline plugin/trino-kinesis/src/test/java/io/trino/plugin/kinesis/TestKinesisConfig.java:47]: Revert unrelated change.

File: plugin/trino-kinesis/src/test/java/io/trino/plugin/kinesis/TestKinesisConfig.java [hunk 4]
```diff
@@ -77,6 +79,7 @@ public void testExplicitPropertyMappings()
                 .setTableDescriptionLocation("/var/lib/kinesis")
                 .setDefaultSchema("kinesis")
                 .setHideInternalColumns(false)
+                .setTableDescriptionRefreshInterval(new Duration(1, TimeUnit.MINUTES))
```
Comment (ebyhr) [inline plugin/trino-kinesis/src/test/java/io/trino/plugin/kinesis/TestKinesisConfig.java:82]: The value in `testExplicitPropertyMappings` should be different from the default value `1m`. 
https://github.com/trinodb/trino/pull/6521#discussion_r552339626

File: plugin/trino-kinesis/src/main/java/io/trino/plugin/kinesis/s3config/S3TableConfigClient.java [hunk 5]
```diff
@@ -96,10 +99,9 @@ public S3TableConfigClient(
     @PostConstruct
     protected void startS3Updates()
     {
-        // TODO: if required make the update interval configurable
         if (this.bucketUrl.isPresent()) {
             ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
-            this.updateTaskHandle = scheduler.scheduleAtFixedRate(this::updateTablesFromS3, 5, 600, TimeUnit.SECONDS);
+            this.updateTaskHandle = scheduler.scheduleAtFixedRate(this::updateTablesFromS3, 5, tableDescriptionRefreshInterval.toMillis(), TimeUnit.MILLISECONDS);
```
Comment (ebyhr) [inline plugin/trino-kinesis/src/main/java/io/trino/plugin/kinesis/s3config/S3TableConfigClient.java:104]: reminder: https://github.com/trinodb/trino/pull/6521#discussion_r552327940

File: docs/src/main/sphinx/connector/kinesis.rst [hunk 6]
```diff
@@ -95,6 +96,14 @@ The S3 bucket and folder will be checked every 10 minutes for updates and change
 
 This property is optional; the default is ``etc/kinesis``.
 
+``kinesis.table-description-refresh-interval``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/kinesis.rst:100]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/src/main/sphinx/connector/kinesis.rst [hunk 7]
```diff
@@ -41,6 +41,7 @@ Property Name                           Description
 ``kinesis.aws-region``                  AWS region to be used to read kinesis stream from
 ``kinesis.default-schema``              Default schema name for tables
 ``kinesis.table-description-location``  Directory containing table description files
+``kinesis.table-description-refresh-interval``  Time for identifying the refresh interval for the directory containing table description files
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/kinesis.rst:44]:
```suggestion
``kinesis.table-description-refresh-interval``  How often to get the table description from S3
```

File: docs/src/main/sphinx/connector/kinesis.rst [hunk 8]
```diff
@@ -95,6 +96,14 @@ The S3 bucket and folder will be checked every 10 minutes for updates and change
 
 This property is optional; the default is ``etc/kinesis``.
 
+``kinesis.table-description-refresh-interval``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+References the refresh interval foor an S3 URL or a folder within Trino deployment that holds one or more JSON files ending with ``.json``, which contain table description files.
+The S3 bucket and folder will be checked every 10 minutes for updates and changed files.
+
+This property is optional; the default is ``1 min``.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/kinesis.rst:105]: This sentence is mostly duplicate of `kinesis.table-description-location` and it's little wrong. How about:
```
This property controls how often the table description is refreshed from S3.

This property is optional; the default is ``1m``.
```

File: docs/src/main/sphinx/connector/kinesis.rst (hunk 7 above)
Comment (ebyhr) [reply docs/src/main/sphinx/connector/kinesis.rst:44]: Please make this table `L36-52` wider. This is one of the CI failure cause. You can build docs by executing `docs/build` locally. If the diff is dirty, I would recommend separating a commit.

**Review by ebyhr — COMMENTED** [review]
Please squash commits into one.

**Comment by somayaj** [comment]
Hi @ebyhr can you squash them from your end? I'm having some merge conflicts here. thanks.

**Comment by ebyhr** [comment]
@somayaj CI failure relates to this change and https://github.com/trinodb/trino/pull/6521#discussion_r552327940 isn't addressed partially. 

I would recommend avoiding to use `master` branch. Please feel free to join community Slack and ping me https://trino.io/slack.html
